Modules/_abc.c

```diff
@@ -81,14 +74,15 @@ static PyObject *
 abc_data_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
     _abc_data *self = (_abc_data *) type->tp_alloc(type, 0);
-    if (self == NULL) {
+    _abcmodule_state *state = PyType_GetModuleState(type);
+    if (self == NULL || state == NULL) {
         return NULL;
```

Comment:
This may leak memory in case ``tp_alloc()`` is succesful but ``PyType_GetModuleState`` fails.

Reply:
Nice catch Thank you :)